src/asm/x86/transform/forward.rs

```diff
@@ -357,21 +357,30 @@ unsafe fn fwd_txfm2d_avx2<T: Coefficient>(
         shift,
       )
     }
+
+    // Avoid zero initialization
+    let tx_in = &mut MaybeUninit::<[MaybeUninit<I32X8>; 64]>::uninit()
+      .assume_init()[..txfm_size_row];
     if cfg.ud_flip {
       // flip upside down
-      for r in 0..txfm_size_row {
-        let input_ptr =
-          input[(txfm_size_row - r - 1) * stride + cg..].as_ptr();
-        tx_in[r] = load_columns(input_ptr, shift);
+      for (in_slice, out_reg) in
+        input[cg..].chunks(stride).zip(tx_in.iter_mut().rev())
+      {
+        *out_reg = MaybeUninit::new(load_columns(in_slice.as_ptr(), shift));
       }
     } else {
-      for r in 0..txfm_size_row {
-        let input_ptr = input[r * stride + cg..].as_ptr();
-        tx_in[r] = load_columns(input_ptr, shift);
+      for (in_slice, out_reg) in
+        input[cg..].chunks(stride).zip(tx_in.iter_mut())
+      {
+        *out_reg = MaybeUninit::new(load_columns(in_slice.as_ptr(), shift));
       }
     }
 
-    txfm_func_col(tx_in, tx_out);
+    // TODO: Don't assume uninitialized values are initialized
+    let tx_out =
+      &mut MaybeUninit::<[I32X8; 64]>::uninit().assume_init()[..txfm_size_row];
+
+    txfm_func_col(assume_slice_init(tx_in), tx_out);
```

Comment:
txfm_func_col could take in the MaybeUninit directly maybe?

Reply:
You are talking about tx_out, right? I thought about it, but didn't want to do it, because I was being lazy. I would have to edit the native implementation and I want to leave that version using a slice.

I'm willing to do it. I just wasn't as willing to when I wrote this originally.

Reply:
txfm_func_col could also just return tx_out, couldn't it? Or is that obnoxious with the simd?

Reply:
I decided to mutate the input and now the patch is a lot bigger than I originally planned.